stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt

```diff
@@ -298,9 +320,15 @@ public class ChannelListViewModel(
      */
     private fun enrichMutedChannels(channels: List<Channel>, mutedChannelIds: List<String>): List<Channel> {
         val mutedChannelIdsSet = mutedChannelIds.toSet()
-        channels.forEach { channel ->
-            channel.isMuted = channel.id in mutedChannelIdsSet
+        return channels.map { channel ->
+            val isMuted = channel.id in mutedChannelIdsSet
+
+            if (channel.isMuted != isMuted) {
+                channel.copy(extraData = channel.extraData.toMutableMap())
```

Comment:
Why do we have to transform the extraData here?

Reply:
1 out of 10 times doesn't work without extra data transformation. Compose just doesn't see the change:
1. Lets imagine we have a channel in the list
2. When mutes are updated we make a shallow copy of the channel with `channel.copy`
3. Both channels reference the very same extra data
4. Updating "muted" field in extra data updates it for both channels
5. As a result, recomposition is not triggered

Reply:
Thanks, that makes sense!